From 07e4a34d175968282e91fc36c26e993e9c4e5228 Mon Sep 17 00:00:00 2001
From: "Gao, HuadongX" <huadongx.gao@intel.com>
Date: Thu, 7 Dec 2023 17:08:58 +0800
Subject: [PATCH] Revert "Prevent NotificationGroup list from updating outside
 of setNotification"

This reverts commit 201eff23c2771c81197666bbdf856c16a4c6ebc2.

Tracked-On: OAM-113161
Signed-off-by: Gao, HuadongX <huadongx.gao@intel.com>
---
 .../car/notification/CarNotificationViewAdapter.java     | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/com/android/car/notification/CarNotificationViewAdapter.java b/src/com/android/car/notification/CarNotificationViewAdapter.java
index a3344d8f..0f070861 100644
--- a/src/com/android/car/notification/CarNotificationViewAdapter.java
+++ b/src/com/android/car/notification/CarNotificationViewAdapter.java
@@ -43,7 +43,6 @@ import com.android.car.ui.recyclerview.ContentLimitingAdapter;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /**
  * Notification data adapter that binds a notification to the corresponding view.
@@ -441,9 +440,9 @@ public class CarNotificationViewAdapter extends ContentLimitingAdapter<RecyclerV
             List<NotificationGroup> unseenNotifications = new ArrayList<>();
             notifications.forEach(notificationGroup -> {
                 if (notificationGroup.isSeen()) {
-                    seenNotifications.add(new NotificationGroup(notificationGroup));
+                    seenNotifications.add(notificationGroup);
                 } else {
-                    unseenNotifications.add(new NotificationGroup(notificationGroup));
+                    unseenNotifications.add(notificationGroup);
                 }
             });
             setSeenAndUnseenNotifications(unseenNotifications, seenNotifications,
@@ -451,9 +450,7 @@ public class CarNotificationViewAdapter extends ContentLimitingAdapter<RecyclerV
             return;
         }
 
-        List<NotificationGroup> notificationGroupList = notifications.stream()
-                .map(notificationGroup -> new NotificationGroup(notificationGroup))
-                .collect(Collectors.toList());
+        List<NotificationGroup> notificationGroupList = new ArrayList<>(notifications);
 
         if (setRecyclerViewListHeadersAndFooters) {
             // add header as the first item of the list.
-- 
2.34.1

